flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/SourceOperator.java

```diff
@@ -254,4 +269,10 @@ private void registerReader() {
 	ListState<SplitT> getReaderState() {
 		return readerState;
 	}
+
+	// Not really meant to be here, this is an example.
+	// TODO: Should we use RuntimeExecutionMode here? I don't like that that one has AUTOMATIC.
+	public enum RuntimeMode {
+		BATCH, STREAMING
```

Comment:
Another option is to use a boolean flag? Will we ever have more modes?

Reply:
I lately try to use more enums than boolean because it's easier to maintain and understand at a glance. Compare `doSomething(13, true, false)` vs. `doSomething(13, STREAMING, EXACTLY_ONCE)`. But yes, here it's just a "custom" boolean. 😅

Reply:
I also think that here we should either go with a boolean as @dawidwys suggested, or with the `RuntimeExecutionMode` that already exists, and not introduce yet another flag.

Reply:
@aljoscha @dawidwys @kl0u I have confusion about which choice should follow. As @aljoscha mentioned that he prefers to use more enums than boolean. And @dawidwys @kl0u prefers to use a boolean. I would like to follow your standard later.

Reply:
Just to be clear. I have no preference here. I just gave another option.